app/subscriber/src/features/search-page/components/advanced-search/AdvancedSearch.tsx

```diff
@@ -61,100 +61,96 @@ export interface IAdvancedSearchProps {
  */
 export const AdvancedSearch: React.FC<IAdvancedSearchProps> = ({ onSearchPage }) => {
   const navigate = useNavigate();
-  const [, { addFilter, getFilter, updateFilter }] = useFilters();
+  const [, { addFilter, updateFilter }] = useFilters();
   const [{ actions }] = useLookup();
-  const [searchParams] = useSearchParams();
   const [
     {
-      search: { filter },
+      search: { filter: search },
     },
-    { storeSearchFilter: storeFilter },
+    { storeSearchFilter },
   ] = useContent();
   const genQuery = useElastic();
   const [{ userInfo }] = useApp();
+  const [{ filter: activeFilter }, { storeFilter }] = useProfileStore();
 
-  const [searchName, setSearchName] = React.useState<string>('');
-  const [viewedFilter, setViewedFilter] = React.useState<IFilterModel>();
+  const [searchName, setSearchName] = React.useState<string>(activeFilter?.name ?? '');
   /** controls the sub group states for media sources. i.e) whether Daily Papers is expanded */
   const [mediaGroupExpandedStates, setMediaGroupExpandedStates] =
     React.useState<ISubMediaGroupExpanded>(defaultSubMediaGroupExpanded);
 
   const isAdmin = userInfo?.roles.includes(Claim.administrator);
-  const filterId = React.useMemo(() => Number(searchParams.get('modify')), [searchParams]);
 
-  const saveSearch = React.useCallback(async () => {
-    const settings = filterFormat(filter, actions);
-    const query = genQuery(settings);
-    await addFilter({
-      name: searchName,
-      query,
-      settings,
-      id: 0,
-      sortOrder: 0,
-      description: '',
-      isEnabled: true,
-      reports: [],
-      folders: [],
-    })
-      .then((data) => toast.success(`${data.name} has successfully been saved.`))
-      .catch();
-  }, [genQuery, filter, actions, addFilter, searchName]);
+  React.useEffect(() => {
```

Comment:
Refactored the add/update logic to keep it simple and clean.  It also now more closely follows how we do it everywhere else in the application.